### ISSUE TITLE
Add tree shaking

### ISSUE THREAD

**Issue by vasilii-kovalev** [epam/UUI#813]
## Description

When importing a single component from the library, the whole library (and some additional stuff) goes to the application's bundle. I compared it with [Ant Design](https://ant.design/) library, and it doesn't add the whole library to the bundle (though it still adds too much), which means it is not the application bundler's problem.

Steps to reproduce:

1. Install a React application using Create React App, as described here: https://create-react-app.dev/docs/getting-started/
  I used the following command: `yarn create react-app test-app --template typescript`
2. Install UUI library as described here: https://uui.epam.com/documents?id=gettingStarted#:~:text=Configuring%20into%20existing%20project (but do not add any library's files yet)
3. Install Ant Design library as described here: https://ant.design/docs/react/introduce#Installation (but do not add any library's files yet)
4. Change/add the following files:
```typescript
// src/index.tsx
import { lazy, StrictMode, Suspense } from "react";
import ReactDOM from "react-dom/client";

import "./index.css";

const AntDesignComponents = lazy(() => import("./ant-design"));
const UUIComponents = lazy(() => import("./uui"));

const root = ReactDOM.createRoot(
  document.getElementById("root") as HTMLElement
);

root.render(
  <StrictMode>
    <Suspense fallback={null}>
      <AntDesignComponents/>
      <UUIComponents/>
    </Suspense>
  </StrictMode>
);
```

```typescript
// src/ant-design.tsx
import { Button } from "antd";
import { FC } from "react";

import "antd/dist/antd.css";

const AntDesignComponents: FC = () => {
  return (
    <Button>Text</Button>
  )
};

export default AntDesignComponents;
```

```typescript
// src/uui.tsx
import { Button } from "@epam/loveship";
import { FC } from "react";

import "@epam/uui-components/styles.css";
import "@epam/loveship/styles.css";

const UUIComponents: FC = () => {
  return (
    <Button caption="Text"/>
  );
};

export default UUIComponents;
```
5. Run the `build` script (`yarn run build` for Yarn)
6. Install `source-map-explorer` to analyze the bundle and add the `analyze` script as described here: https://create-react-app.dev/docs/analyzing-the-bundle-size/
7. Run the `analyze` script (`yarn run analyze` for Yarn)
8. Check the generated report

The whole bundle:
![image](https://user-images.githubusercontent.com/10310491/200028643-100d9e68-2cad-4eac-bc99-cad3a4394605.png)

UUI library chunk:
![image](https://user-images.githubusercontent.com/10310491/200029560-8acff829-17cc-46d0-82f6-b547d5f8bd9b.png)

Ant Design chunk:
![image](https://user-images.githubusercontent.com/10310491/200029644-f5565289-ddf7-4cdd-afa9-c663789af636.png)

As you can see, importing a single button component resulted in about 823 KB of bundled code, including all the components, some of Lodash functions, Day.js, etc.

The problem is reproducible for React 17 and 18, and `react-script` package's version 5.0.1.

## Describe the solution you'd like

1. I want to have only code I use bundled. Would be great, if the tree shaking would happen automatically, but it is also acceptable to have other options. For example, individual imports [like in Lodash](https://lodash.com/per-method-packages) (`import isUndefined from "lodash/isUndefined"`)
2. In addition, it would also be useful to add tree shaking for CSS files, or at least provide a tip about how to do that on the consumer's side
3. (If possible) add some checks to track the tree shaking state, to make sure it won't break in the future

**Comment by AlekseyManetov** [comment]
HI, we already working on our build toolchain rework and one of the main points of it - tree shaking support, hope we can make it without any breaking changes in imports or something else, it will work out of the box, if your own bundler can exclude code which is not used from result bundle.

**Comment by NatalliaAlieva** [comment]
Released in 5.0.0 ver.
